assertion/function/assertiontree/assertion_node.go

```diff
@@ -93,7 +93,7 @@ func (n *assertionNodeCommon) SetConsumeTriggers(triggers []*annotation.ConsumeT
 
 // nilable(node, result 0)
 func (n *assertionNodeCommon) Root() *RootAssertionNode {
-	if n == nil {
+	if n == nil || n.parent == nil {
```

Comment:
(fix for NilAway reported error that was reported due the support added in this PR)

Reply:
> Wait....isn't the original logic incorrect? we would recursively go to n.parent until n == nil, and then return a nil instead. So Root() always return nil in all cases.

You are right that this function would result in recursively going up to the root of the assertion tree, which is of type `RootAssertionNode`. However, `RootAssertionNode` overrides the `Root()` method to return itself, which is why `nil` was never being returned from here.

> A quick search reveals that this function isn't called anywhere, can we simply remove this function then?

I actually see it being used from a lot of places, such as
[https://sourcegraph.uberinternal.com/code.uber.internal/go-code/-/blob/src/code.uber.internal/infra/progsys/nilaway/assertion/function/assertiontree/var_assertion_node.go#L53](https://sourcegraph.uberinternal.com/code.uber.internal/go-code/-/blob/src/code.uber.internal/infra/progsys/nilaway/assertion/function/assertiontree/var_assertion_node.go#L53)
[https://sourcegraph.uberinternal.com/code.uber.internal/go-code/-/blob/src/code.uber.internal/infra/progsys/nilaway/assertion/function/assertiontree/var_assertion_node.go#L71](https://sourcegraph.uberinternal.com/code.uber.internal/go-code/-/blob/src/code.uber.internal/infra/progsys/nilaway/assertion/function/assertiontree/var_assertion_node.go#L71)